### PR TITLE
[cpu]freebsd: cpu.Info() now returns CPUInfos same number as CPU nums.

### LINKED ISSUE #304: Mhz copied to CPU on FreeBSD
I noticed CPU outputs value of MHz instead of number on CPUs. I thought CPU is used to output number of physical CPUs.

I think dmesg might help here to get the right number. Here are a few examples from our servers

- (2phys CPUs no HT) FreeBSD/SMP: 2 package(s) x 4 core(s)  `hw.ncpu=8`
- (1 phys CPUs) FreeBSD/SMP: 1 package(s) x 2 core(s) x 2 hardware threads  `hw.ncpu=4`
- (1 phys CPUs) FreeBSD/SMP: 1 package(s) x 4 core(s) x 2 SMT threads  `hw.ncpu=8`
- (1 phys CPUs) FreeBSD/SMP: 1 package(s) x 6 core(s) x 2 SMT threads  `hw.ncpu=12`

### PR DESCRIPTION
**Pull request by shirou**
fix #304

### CODE REVIEW

**Comment by tmm1** [comment]
This PR broke `cpu.Info()` on freebsd for me.. now it only returns an empty array. Here's my dmesg.boot:

```
CPU: Intel(R) Core(TM) i5-2415M CPU @ 2.30GHz (2294.84-MHz K8-class CPU)
  Origin="GenuineIntel"  Id=0x206a7  Family=0x6  Model=0x2a  Stepping=7
  Features=0x783fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,MMX,FXSR,SSE,SSE2>
  Features2=0x9e98220b<SSE3,PCLMULQDQ,MON,SSSE3,CX16,SSE4.1,SSE4.2,POPCNT,AESNI,XSAVE,OSXSAVE,AVX,HV>
  AMD Features=0x28100800<SYSCALL,NX,RDTSCP,LM>
  AMD Features2=0x1<LAHF>
  TSC: P-state invariant
```

**Comment by tmm1** [comment]
I am using 10.3-RELEASE-p11 and there is no such line in my dmesg.boot like "FreeBSD/SMP"

File: cpu/cpu_freebsd.go [hunk 1]
```diff
@@ -133,19 +135,28 @@ func Info() ([]InfoStat, error) {
...
+		ret[i] = c
+	}
+
+	return ret, nil
+}
+
+func parseDmesgBoot(fileName string) (InfoStat, int, error) {
+	c := InfoStat{}
+	lines, _ := common.ReadLines(fileName)
+	var cpuNum int
```
Comment (tmm1) [inline cpu/cpu_freebsd.go]: Here I think cpuNum should default to 1

Comment (shirou) [inline cpu/cpu_freebsd.go]: Hmm, I see. Changing to `cpuNum := 1` solves on your environment?

Comment (tmm1) [inline cpu/cpu_freebsd.go]: Yes, I just tried it and that fix works.

Comment (shirou) [inline cpu/cpu_freebsd.go]: Thanks! I will fix on the master branch and push very soon.

Comment (shirou) [inline cpu/cpu_freebsd.go]: just pushed b6da2bd76e7d66a228181f7993265cf859d5351f . Thank you for your information!

Comment (tmm1) [inline cpu/cpu_freebsd.go]: Thanks for the quick fix!

**Pull request merged**